Capturing like/llike functions in /docs/reference/operations/regex.md #1078 (#1103)
questdb/questdb.io: docs/reference/operators/regex.md

`string !~ regex` - checks if `string` value does not match regex

`symbol !~ regex` - checks if `symbol` value does not match regex

## LIKE 

> The `LIKE` operator returns true if it matches the specified pattern

Reviewer: 1. Please follow [the template](https://github.com/questdb/questdb.io/blob/master/docs/__guidelines/template/function.md#function_name) to document this:
 
'LIKE' - returns...
 This applies to the rest of the PR.
2. I think it is not just returns true but returns the row?

Author: The operator itself returns a boolean but the WHERE clause returns the row if the operator result is true

![Screenshot from 2022-10-05 15-55-46](https://user-images.githubusercontent.com/36854536/194041163-484b77fb-7d1d-4a24-bf10-7777a6e40d20.png)

`string !~ regex` - checks if `string` value does not match regex

`symbol !~ regex` - checks if `symbol` value does not match regex

## LIKE / ILIKE

`string LIKE pattern` - returns true if `string` value matches `pattern` otherwise returns false (case sensitive match)

`string ILIKE pattern` - returns true if `string` value matches `pattern` otherwise returns false (case-insensitive match)

### Arguments
`string` is [string].

> `symbol` is [string].

Author: @amyshwang correct me if I'm wrong, but I don't think "pattern following the Java regex syntax" is an appropriate description for pattern here since the pattern does not support the whole range of regex meta characters rather it supports only the two wildcards '?' and '%'.

> `symbol` is [string].

Reviewer: Yes you're right! Apologies for the confusion. But I think it should be "_" and "%"?

> `symbol` is [string].

Author: Yep my bad. It's '_' and not '?'

### Return value

Return value type is [boolean].

### Description

> The pattern can contain wildcards which are interpreted as follows:

Reviewer: ```suggestion
The Java regex syntax can contain wildcards which are interpreted as follows:
```

- `?` - matches any single character
- `%` - matches any sequence of zero or more characters

Wildcards can be used as follows:

|      query             | result  |
| ---------------------- | ------- |

> | 'quest' LIKE 'quest'   |  true   |

Reviewer: Applying this to all:
```suggestion
|  SELECT 'quest' LIKE 'quest'   |  true   |
```

In this way, users can copy it and run it on the demo site directly.

`string !~ regex` - checks if `string` value does not match regex

`symbol !~ regex` - checks if `symbol` value does not match regex

## LIKE / ILIKE

`(string) LIKE (pattern)` - returns true if the `string` value matches `pattern`, otherwise returns false (case sensitive match).

`(string) ILIKE (pattern)` - returns true if the `string` value matches `pattern`, otherwise returns false (case-insensitive match).

### Arguments

> `string` is a string which is commonly the name of a column.

Reviewer: ```suggestion
`string` is an expression that evaluates to the `string` data type. 
```

`pattern` is a pattern which can contain wildcards like `?` and `%`.

### Return value

Return value type is `boolean`.

### Description

If the pattern doesn't contain wildcards, then the pattern represents the string itself.

The wildcards whuch can be used in pattern are interpreted as follows:
- `?` - matches any single character

> - `%` - matches any sequence of zero or more characters

Reviewer: Missing full stops.

Wildcards can be used as follows:

|            query              | result  |
| ----------------------------- | ------- |
| SELECT 'quest' LIKE 'quest'   |  true   |
| SELECT 'quest' LIKE 'quest_'  |  true   |
| SELECT 'quest' LIKE 'que%'    |  true   |
| SELECT 'quest' LIKE '\_ues_'  |  true   |
| SELECT 'quest' LIKE 'q_'      |  false  |

`ILIKE` performs a case-insensitive match as follows:

|            query              | result  |
| ----------------------------- | ------- |
| SELECT 'quest' ILIKE 'QUEST'  |  true   |
| SELECT 'qUeSt' ILIKE 'QUEST'  |  true   |
| SELECT 'quest' ILIKE 'QUE%'   |  true   |
| SELECT 'QUEST' ILIKE '\_ues_' |  true   |

> ### Example

Reviewer: ```suggestion
### Examples
```

#### LIKE

```questdb-sql 
SELECT * FROM trades
WHERE symbol LIKE '%-USD'
LATEST ON timestamp PARTITION BY symbol;
```

| symbol | side | price | amount | timestamp |
| --- | --- | --- | --- | --- |
| ETH-USD | sell | 1348.13 | 3.22455108 | 2022-10-04T15:25:58.834362Z |
| BTC-USD | sell | 20082.08 | 0.16591219 | 2022-10-04T15:25:59.742552Z |

#### ILIKE

```questdb-sql 
SELECT * FROM trades
WHERE symbol ILIKE '%-usd'
LATEST ON timestamp PARTITION BY symbol;
```

> | symbol | side | price | amount | timestamp |

Reviewer: As mentioned, we'd like proper behaviours to be documented so this is not correct. In addition, https://github.com/questdb/questdb/issues/2585 has been merged.

Author: By proper behaviour do you mean:
1) the result it returns on the demo site (which is still not updated with the correct ILIKE implementation).
2) the assumed correct behaviour for the ILIKE operator.

In my PR, I have used the assumed behaviour.

Reviewer: You're right! My bad!

`symbol !~ regex` - checks if `symbol` value does not match regex

## LIKE / ILIKE

`(string) LIKE (pattern)` - returns true if the `string` value matches `pattern`, otherwise returns false (case sensitive match).

`(string) ILIKE (pattern)` - returns true if the `string` value matches `pattern`, otherwise returns false (case-insensitive match).

### Arguments

`string` is an expression that evaluates to the `string` data type.

> `pattern` is a pattern which can contain wildcards like `?` and `%`.

Author: Yep made a mistake. Sorry for the confusion.

### Return value

Return value type is `boolean`.

### Description

If the pattern doesn't contain wildcards, then the pattern represents the string itself.

> 

Reviewer 2: typo: `whuch`

The wildcards whuch can be used in pattern are interpreted as follows:
- `?` - matches any single character.
- `%` - matches any sequence of zero or more characters.

Wildcards can be used as follows:

|            query              | result  |
| ----------------------------- | ------- |

> | SELECT 'quest' LIKE 'quest'   |  true   |

Reviewer 2: Does : 
```sql
SELECT 'quest' LIKE 'quest_'
```
really return true ? 

It doesn't seem to be the case on master branch or demo instance .

Author: Yep my bad, 

it's supposed to be SELECT 'quest' LIKE 'ques_'

Author: As for the second one, 
I intended it to be: SELECT 'quest' LIKE '_ues_'
I added the '\' to escape the underscore as it was displaying it in italics.

![Screenshot from 2022-10-11 15-22-46](https://user-images.githubusercontent.com/36854536/195058681-c87c1517-02ce-41f6-bb4f-c2aae58e3c72.png)

Reviewer: Could you please revert the change? We need to check the rendering, not the source!

Author: yep done
